### PR TITLE
TNO-136 Fix hibernate concurrency

### PR DESCRIPTION
**Pull request by Fosol**
Because Java is horrible and can't handle date values or serialization.  I have added a new `version` column to each table for optimistic concurrency.

I've updated the various library versions to support this.

> This requires a `make db-refresh`
> This requires a `make refresh n=api-editor`

### CODE REVIEW

File: api/editor/api-editor/pom.xml [hunk 1]
```diff
@@ -52,10 +52,11 @@
   </repositories>
 
   <dependencies>
+
     <dependency>
       <groupId>ca.bc.gov.tno</groupId>
       <artifactId>dal-db</artifactId>
-      <version>0.0.19-SNAPSHOT</version>
+      <version>0.0.20-SNAPSHOT</version>
```
Comment (Fosol) [inline api/editor/api-editor/pom.xml:59]: Bump version

File: api/editor/api-editor/src/main/java/ca/bc/gov/tno/controllers/HomeController.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
 @RestController
 public class HomeController {
 
-  @GetMapping(path = { "", "/" })
+  @GetMapping(path = { "", "/", "/api", "/api/" })
```
Comment (Fosol) [inline api/editor/api-editor/src/main/java/ca/bc/gov/tno/controllers/HomeController.java:17]: Fix issues Jonny was running into with Postman

File: libs/java/dal/db/dal-db-migration/pom.xml [hunk 3]
```diff
@@ -65,7 +65,7 @@
     <dependency>
       <groupId>ca.bc.gov.tno</groupId>
       <artifactId>dal-db</artifactId>
-      <version>0.0.18-SNAPSHOT</version>
+      <version>0.0.20-SNAPSHOT</version>
```
Comment (Fosol) [inline libs/java/dal/db/dal-db-migration/pom.xml:68]: Bump version

File: libs/java/dal/db/dal-db-test/pom.xml [hunk 4]
```diff
@@ -29,7 +29,7 @@
     <dependency>
       <groupId>ca.bc.gov.tno</groupId>
       <artifactId>dal-db</artifactId>
-      <version>0.0.18-SNAPSHOT</version>
+      <version>0.0.20-SNAPSHOT</version>
```
Comment (Fosol) [inline libs/java/dal/db/dal-db-test/pom.xml:32]: Bump version

File: libs/java/dal/db/dal-db/pom.xml [hunk 5]
```diff
@@ -6,7 +6,7 @@
 
   <groupId>ca.bc.gov.tno</groupId>
   <artifactId>dal-db</artifactId>
-  <version>0.0.19-SNAPSHOT</version>
+  <version>0.0.20-SNAPSHOT</version>
```
Comment (Fosol) [inline libs/java/dal/db/dal-db/pom.xml:9]: Bump version

File: libs/java/dal/db/dal-db/src/main/java/ca/bc/gov/tno/dal/db/entities/ContentReferencePK.java [hunk 6]
```diff
@@ -93,7 +93,7 @@ public boolean equals(Object obj) {
   @Override
   public String toString() {
     final StringBuilder sb = new StringBuilder("{");
-    sb.append("source=").append(source);
+    sb.append("source='").append(source).append("\'");
```
Comment (Fosol) [inline libs/java/dal/db/dal-db/src/main/java/ca/bc/gov/tno/dal/db/entities/ContentReferencePK.java:96]: Fix serialization

File: libs/java/dal/db/dal-db/src/main/java/ca/bc/gov/tno/dal/db/entities/DataSourceSchedulePK.java [hunk 7]
```diff
@@ -68,7 +68,7 @@ public boolean equals(Object obj) {
   public String toString() {
     final StringBuilder sb = new StringBuilder("{");
     sb.append("dataSourceId=").append(dataSourceId);
-    sb.append(", scheduleId='").append(scheduleId).append("'");
+    sb.append(", scheduleId=").append(scheduleId);
```
Comment (Fosol) [inline libs/java/dal/db/dal-db/src/main/java/ca/bc/gov/tno/dal/db/entities/DataSourceSchedulePK.java:71]: Fix serialization

File: libs/java/dal/db/dal-db/src/main/java/ca/bc/gov/tno/dal/db/entities/TimeTrackingPK.java [hunk 8]
```diff
@@ -66,7 +66,7 @@ public boolean equals(Object obj) {
   public String toString() {
     final StringBuilder sb = new StringBuilder("{");
     sb.append("contentId=").append(contentId);
-    sb.append(", uid=").append(userId);
+    sb.append(", userId=").append(userId);
```
Comment (Fosol) [inline libs/java/dal/db/dal-db/src/main/java/ca/bc/gov/tno/dal/db/entities/TimeTrackingPK.java:69]: Fix serialization

File: libs/java/service/pom.xml [hunk 9]
```diff
@@ -6,7 +6,7 @@
 
   <groupId>ca.bc.gov.tno</groupId>
   <artifactId>service</artifactId>
-  <version>0.0.8-SNAPSHOT</version>
+  <version>0.0.9-SNAPSHOT</version>
```
Comment (Fosol) [inline libs/java/service/pom.xml:9]: Bump version

File: openshift/templates/services/syndication/deploy.yaml [hunk 10]
```diff
@@ -253,26 +253,26 @@ objects:
...
-                httpGet:
-                  path: '/health'
-                  port: ${{APP_PORT}}
-                  scheme: HTTP
-                initialDelaySeconds: 120
-                timeoutSeconds: 60
-                periodSeconds: 30
-                successThreshold: 1
-                failureThreshold: 3
+              # livenessProbe:
```
Comment (Fosol) [inline openshift/templates/services/syndication/deploy.yaml:256]: If I add this back I have to change the delay time to 5-10 minutes!  It's ridiculous.  Java starts up so incredibly slow.

File: services/syndication/pom.xml [hunk 11]
```diff
@@ -74,7 +74,7 @@
     <dependency>
       <groupId>ca.bc.gov.tno</groupId>
       <artifactId>service</artifactId>
-      <version>0.0.8-SNAPSHOT</version>
+      <version>0.0.9-SNAPSHOT</version>
```
Comment (Fosol) [inline services/syndication/pom.xml:77]: Bump version

File: api/editor/api-editor/src/main/java/ca/bc/gov/tno/controllers/HomeController.java (hunk 2 above)
Comment (jdtoombs) [reply api/editor/api-editor/src/main/java/ca/bc/gov/tno/controllers/HomeController.java:17]: woo!

**Review by jdtoombs — APPROVED** [review]

**Pull request merged**